File: Proyecto-Prog_Avanzada_Frontend-master/src/app/modelo/detalle-compra-dto.ts
import { ProductoDTO } from "./producto-dto";

export class DetalleCompraDTO {
    //producto: ProductoDTO = new ProductoDTO();
    codigoPublicacion: number = 0;
    unidades: number = 0;
    valorTotal: number = 0;
    precioUnidad: number = 0.0;

    constructor(codigoPublicacion: number, unidades: number) {
        //this.producto= producto;
        this.codigoPublicacion = codigoPublicacion;
        this.unidades = unidades;
        //this.valorTotal = unidades * producto.precio;
    }
}

File: Proyecto-Prog_Avanzada-railway/src/main/java/co/edu/uniquindio/proyecto/proyectotienda/dto/DetalleCompraDTO.java
package co.edu.uniquindio.proyecto.proyectotienda.dto;

import lombok.*;

@AllArgsConstructor
@Getter
@Setter
@ToString
public class DetalleCompraDTO {
    private int codigoPublicacion;
    private int unidades;
    private double precioUnidad;
}
